gtwrap/pybind_wrapper.py

```diff
@@ -254,37 +254,100 @@ def wrap_operators(self, operators, cpp_class, prefix='\n' + ' ' * 8):
                     op.operator))
         return res
 
+    def wrap_enum(self, enum, class_name='', module=None, prefix=' ' * 4):
+        """
+        Wrap an enum.
+
+        Args:
+            enum: The parsed enum to wrap.
+            class_name: The class under which the enum is defined.
+            prefix: The amount of indentation.
+        """
+        if module is None:
+            module = self._gen_module_var(enum.namespaces())
+
+        cpp_class = enum.cpp_typename().to_cpp()
+        if class_name:
+            # If class_name is provided, add that as the namespace
+            cpp_class = class_name + "::" + cpp_class
+
+        res = '{prefix}py::enum_<{cpp_class}>({module}, "{enum.name}", py::arithmetic())'.format(
+            prefix=prefix, module=module, enum=enum, cpp_class=cpp_class)
+        for enumerator in enum.enumerators:
+            res += '\n{prefix}    .value("{enumerator.name}", {cpp_class}::{enumerator.name})'.format(
+                prefix=prefix, enumerator=enumerator, cpp_class=cpp_class)
+        res += ";\n\n"
+        return res
+
+    def wrap_enums(self, enums, instantiated_class, prefix=' ' * 4):
+        """Wrap multiple enums defined in a class."""
+        cpp_class = instantiated_class.cpp_class()
+        module_var = instantiated_class.name.lower()
+        res = ''
+
+        for enum in enums:
```

Comment:
just double checking, is there any chance of this throwing a "Nonetype is not iterable" error?
i.e. hopefully the default value of enums is `[]` rather than `None`?

Reply:
In `Class` in `classes.py` we have `self.enums = []` so all good :slightly_smiling_face: